src/wallet/wallet.cpp

```diff
@@ -690,6 +690,11 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         // bits of the unencrypted private key in slack space in the database file.
         dbw->Rewrite();
 
+        // BDB seems to have a bad habit of writing old data into
```

Comment:
In commit "After encrypting the wallet, reload the database environment"

Delayed flushing seems more like a legitimate design tradeoff than a bad habit. Maybe just say something like "flush and reload the database environment here to clear out any data in memory that might be left behind after the rewrite above."

Reply:
The comment was just copied over from `rpcwallet.cpp`.